src/test/java/com/adyen/StoredValueTest.java

```diff
@@ -0,0 +1,182 @@
+import com.adyen.model.storedvalue.StoredValueLoadRequest;
+import com.adyen.model.storedvalue.StoredValueLoadResponse;
+import com.adyen.model.storedvalue.StoredValueStatusChangeRequest;
+import com.adyen.model.storedvalue.StoredValueStatusChangeResponse;
+import com.adyen.model.storedvalue.StoredValueVoidRequest;
+import com.adyen.model.storedvalue.StoredValueVoidResponse;
+import com.adyen.service.StoredValue;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests for
+ * /issue
+ * /changeStatus
+ * /load
+ * /checkBalance
+ * /mergeBalance
+ * /voidTransaction
+ */
+public class StoredValueTest extends BaseTest {
+
+
+    @Test
+    public void testIssueSuccessMocked() throws Exception {
+        Client client = createMockClientFromFile("mocks/storedvalue/issue-success.json");
+        StoredValue storedValue = new StoredValue(client);
+
+        StoredValueIssueRequest storedValueIssueRequest = new StoredValueIssueRequest();
+        storedValueIssueRequest.setMerchantAccount("merchantAccount");
+        storedValueIssueRequest.setReference("issueReference");
+        storedValueIssueRequest.setPaymentMethod(new DefaultPaymentMethodDetails().type("givex"));
+
+        Amount amount = new Amount();
+        amount.setCurrency("EUR");
+        amount.setValue(1000L);
+
+        storedValueIssueRequest.setAmount(amount);
+
+        StoredValueIssueResponse storedValueIssueResponse = storedValue.issue(storedValueIssueRequest);
+        assertEquals("EUR", storedValueIssueResponse.getCurrentBalance().getCurrency());
+        assertEquals(new Long(1000), storedValueIssueResponse.getCurrentBalance().getValue());
+        assertEquals("851591622730873A", storedValueIssueResponse.getPspReference());
+        assertEquals(StoredValueIssueResponse.ResultCodeEnum.SUCCESS, storedValueIssueResponse.getResultCode());
+        assertEquals("6036280000000000000", storedValueIssueResponse.getPaymentMethod().get("number"));
+        assertEquals("givex", storedValueIssueResponse.getPaymentMethod().get("type"));
+    }
+
+    @Test
+    public void testChangeStatusActiveSuccessMocked() throws Exception {
+        Client client = createMockClientFromFile("mocks/storedvalue/change-status-active-success.json");
+        StoredValue storedValue = new StoredValue(client);
+
+        StoredValueStatusChangeRequest storedValueStatusChangeRequest = new StoredValueStatusChangeRequest();
+        storedValueStatusChangeRequest.setStatus(StoredValueStatusChangeRequest.StatusEnum.ACTIVE);
+
+        Amount amount = new Amount();
+        amount.setCurrency("EUR");
+        amount.setValue(1000L);
+
+        storedValueStatusChangeRequest.setAmount(amount);
+        storedValueStatusChangeRequest.setMerchantAccount("merchantAccount");
+        storedValueStatusChangeRequest.setPaymentMethod(new DefaultPaymentMethodDetails().type("givex").number("6036280000000000000").securityCode("1111"));
```

Comment:
shall we introduce a givex payment method ? extending from a genericGiftCard payment method ?